staging/src/k8s.io/apiserver/pkg/apis/flowcontrol/bootstrap/default.go

```diff
@@ -0,0 +1,336 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bootstrap
+
+import (
+	coordinationv1 "k8s.io/api/coordination/v1"
+	corev1 "k8s.io/api/core/v1"
+	flowcontrol "k8s.io/api/flowcontrol/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/serviceaccount"
+	"k8s.io/apiserver/pkg/authentication/user"
+)
+
+// The objects that define an apiserver's initial behavior
+var (
+	PreservedPriorityLevelConfigurations = []*flowcontrol.PriorityLevelConfiguration{
+		DefaultPriorityLevelConfigurationExempt,
+		DefaultPriorityLevelConfigurationDefault,
+	}
+	PreservedFlowSchemas = []*flowcontrol.FlowSchema{
+		DefaultFlowSchemaExempt,
+		DefaultFlowSchemaDefault,
+	}
+)
+
+// Initial PriorityLevelConfiguration objects
+var (
+	DefaultPriorityLevelConfigurationExempt = newPriorityLevelConfiguration(
+		flowcontrol.PriorityLevelConfigurationNameExempt,
+		flowcontrol.PriorityLevelConfigurationSpec{
+			Type: flowcontrol.PriorityLevelEnablementExempt,
+		},
+	)
+	DefaultPriorityLevelConfigurationDefault = newPriorityLevelConfiguration(
+		"default", // the "catch-all" priority-level
+		flowcontrol.PriorityLevelConfigurationSpec{
+			Type: flowcontrol.PriorityLevelEnablementLimited,
+			Limited: &flowcontrol.LimitedPriorityLevelConfiguration{
+				AssuredConcurrencyShares: 100,
+				LimitResponse: flowcontrol.LimitResponse{
+					Type: flowcontrol.LimitResponseTypeQueue,
+					Queuing: &flowcontrol.QueuingConfiguration{
+						Queues:           128,
+						HandSize:         6,
+						QueueLengthLimit: 100,
+					},
+				},
+			},
+		})
+)
+
+// Initial FlowSchema objects
+var (
+	DefaultFlowSchemaExempt = newFlowSchema(
+		"exempt",
+		flowcontrol.PriorityLevelConfigurationNameExempt,
+		1000, // matchingPrecedence
+		"",   // distinguisherMethodType
+		flowcontrol.PolicyRulesWithSubjects{
+			Subjects: groups(user.SystemPrivilegedGroup),
+			ResourceRules: []flowcontrol.ResourcePolicyRule{
+				resourceRule(
+					[]string{flowcontrol.VerbAll},
+					[]string{flowcontrol.APIGroupAll},
+					[]string{flowcontrol.ResourceAll},
+					[]string{flowcontrol.NamespaceEvery},
+					true,
+				),
+			},
+			NonResourceRules: []flowcontrol.NonResourcePolicyRule{
+				nonResourceRule(
+					[]string{flowcontrol.VerbAll},
+					[]string{flowcontrol.NonResourceAll},
+				),
+			},
+		},
+	)
+	DefaultFlowSchemaDefault = newFlowSchema(
+		"default",
+		"default",
+		10000, // matchingPrecedence
```

Comment:
I think the catch-all FlowSchema should have the logically lowest (numerically highest) possible matching precedence.  No FlowSchema with a logically lower matching precedence will have any effect.  We should make it impossible to create such a necessarily ineffective FlowSchema, and we should leave as much of the matching precedence value range available to administrators as possible.

Reply:
or can we constraint the precedence to range [1000,10000] by the validation? it's a bit odd to see some confusing number like `(1<<32)-1` in an object.

Reply:
We discussed this value in our last meeting and agreed on using the max `int32`.  Any smaller value creates a booby trap for admins: they can create FlowSchema objects that are guaranteed to have no effect.  Also, this number appears only in a pre-created object --- admins do not have to put it in any of their objeccts.  Also, this number is used elsewhere in the kube API for a similar purpose.

Reply:
Yeah, since this will match literally every request, we need to make sure that it loses all ties. So, this needs to be the lowest value permitted by validation, whatever that is.

Note that `(1<<32)-1` is not a positive 32 bit int. ;)

I'm fine with changing validation to make 10000 the max. I would not make 1000 the min though.

Reply:
i refactored the model to constraint the range of matching-precedence to [0,10000] by https://github.com/kubernetes/kubernetes/pull/85841